cylc/flow/scripts/lint.py

```diff
@@ -0,0 +1,162 @@
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""Look through a folder for Cylc 7 syntax ``suite*.rc`` files
+"""
+from optparse import Values
+from pathlib import Path
+import re
+from typing import List
+
+from cylc.flow import LOG
+from cylc.flow.option_parsers import (
+    CylcOptionParser as COP,
+)
+from cylc.flow.terminal import cli_function
+
+
+FILEGLOBS = ['*.rc']
+CHECK78 = {
+    '7-to-8': {
+        re.compile(r'\[cylc\]'): {
+            'short': 'section `[cylc]` is now called `[scheduler]`.',
+            'url': 'major-changes/config-changes.html#graph'
+        },
+        re.compile(r'include at start-up\s?='): {
+            'short': '`include at start up` is obselete.',
+            'url': 'major-changes/config-changes.html#graph'
+        },
+        re.compile(r'exclude at start-up\s?='): {
+            'short': '`exclude at start up` is obselete.',
+            'url': 'major-changes/config-changes.html#graph'
+        },
+        re.compile(r'log resolved dependencies\s?='): {
+            'short': '`exclude at start up` is obselete.',
+            'url': 'major-changes/config-changes.html#graph'
+        },
+        re.compile(r'\[\[dependencies\]\]'): {
+            'short': '`[dependencies]` is deprecated.',
+            'url': 'major-changes/config-changes.html#graph'
+        },
+        re.compile(r'graph\s?='): {
+            'short': (
+                '`[cycle point]graph =` is deprecated, '
+                'use `cycle point = <graph>`',
+            ),
+            'url': 'major-changes/config-changes.html#graph'
+        },
+        re.compile(r'\[\[remote\]\]'): {
+            'short': (
+                '`[runtime][<namespace>][remote]host` is deprecated, '
+                'use `[runtime][<namespace>]platform`',
+            ),
+            'url': 'major-changes/config-changes.html#graph'
+        },
+        re.compile(r'\[\[job\]\]'): {
+            'short': (
+                '`[runtime][<namespace>][job]` is deprecated, '
+                'use `[runtime][<namespace>]platform`',
+            ),
+            'url': 'major-changes/config-changes.html#graph'
+        }
+    }
+}
+
+
+def parse_checks():
+    """Collapse metadata in checks dicts.
+    """
+    parsedchecks = {}
+    index = 0
+    for checkdict in [CHECK78]:
+        for purpose, checks in checkdict.items():
+            for pattern, meta in checks.items():
+                meta.update({'purpose': purpose})
+                meta.update({'index': index})
+                parsedchecks.update({pattern: meta})
+                index += 1   # noqa SIM113
+    return parsedchecks
+
+
+def check_cylc_file(file_, checks, modify=False):
+    """Check A Cylc File for Cylc 7 Config"""
+    # Set mode as read-write or read only.
+    outlines = []
+
+    # Open file, and read it's line to mempory.
+    lines = file_.read_text().split('\n')
+    count = 0
+    for line_no, line in enumerate(lines):
```

Comment:
Having looked at this I'm not so sure that this is a good idea - I think that it could get very complex very quickly and hard to determine line numbers.